core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -109,6 +110,22 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
     if (destination.id !in topLevelDestinations) {
       handleDrawerOnNavigation()
     }
+    if (destination.id == readerFragmentResId) {
+      unLockReaderNavView()
+    } else {
+      lockReaderNavView()
+    }
+  }
+
+  private fun unLockReaderNavView() {
+    drawerContainerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED, readerDrawerNavView)
```

Comment:
also we could probably get rid of these method with an extension function on `NavigationView` in this class so it more goes

`readerDrawerNavView.setLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED)`

and then the code is speaking clearly